lightning/restore/restore.go

```diff
@@ -814,7 +825,31 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
 	}
 
 	wg.Wait()
-	close(stopPeriodicActions)
+	// if context is done, should return directly
+	select {
+	case <-ctx.Done():
+		err = restoreErr.Get()
+		if err == nil {
+			err = ctx.Err()
+		}
+		logTask.End(zap.ErrorLevel, err)
+		return err
+	default:
+	}
+
+	close(postProcessTaskChan)
+	// otherwise, we should run all tasks in the post-process task chan
+	for i := 0; i < rc.cfg.App.TableConcurrency; i++ {
+		wg.Add(1)
+		go func() {
+			for task := range postProcessTaskChan {
+				_, err := task.tr.postProcess(ctx, rc, task.cp, true)
```

Comment:
I think the last parameter means "can't delay".  if it's "can delay", I expect that this function may or may not run something in this turn.

Reply:
replaced with a clearer name